File: src/main/java/jisop/command_line/parse/OptionParameter.java
package jisop.command_line.parse;

import jisop.infrastructure.Strings;

/**
 *
 * @author mathieu
 */
class OptionParameter {
       public static ArgumentParameter tryParse(String value)
        {
            if (value.contains("|"))
            {
                String[] names = Strings.split(Strings.trimEnd(value, "=:"), '|');
                String delimiter = null;
                String last = value.substring(value.length()-2, value.length()-1);
                switch (last.charAt(0))
                {
                    case '=':
                    case ':':
                        delimiter = last;
                        break;
                    default:
                        break;
                }
                return new ArgumentParameter(value, names, delimiter, null);
            }
            return null;
        }
}

File: src/main/java/isop/command_line/parse/OptionParameter.kt
package isop.command_line.parse

import isop.infrastructure.Strings

/**
 *
 * @author mathieu
 */
internal object OptionParameter {
    fun tryParse(value: String): ArgumentParameter? {
        if (value.contains("|")) {
            val names = Strings.split(Strings.trimEnd(value, "=:"), '|')
            var delimiter: String? = null
            val last = value.substring(value.length - 2, value.length - 1)
            when (last[0]) {
                '=', ':' -> delimiter = last
                else -> {
                }
            }
            return ArgumentParameter(value, names, delimiter, null)
        }
        return null
    }
}
